### ISSUE TITLE
"Just listening" mode for video chats

### ISSUE THREAD

**Issue by bcpierce00** [clowdr-app/clowdr-web-app#53]
One way to lower the "activation threshold" for people to join conversations is to allow them to join in "just listening" mode.  

E.g., we could make this the default when people enter a room.  They are able to hear (and see?) what's going on, but other people cannot see or hear them -- their name is displayed in a separate part of the UI for people in the room (so that they know who is listening but it's clear that they are not really part of the conversation, don't wish to be asked questions or invited to speak, etc.). 

When in listening mode, there is a clear indication that this is what is happening and a prominent button that says "Enter room."  (And another for "leave room.)

**Comment by bcpierce00** [comment]
We're out of time for ICFP, but I still want to try experimenting with this.

**Comment by wouter-swierstra** [comment]
+1

Entering a room (with the camera on) can be quite disruptive to the people there. Audio only when entering to listen to the conversation and turning on your camera to participate may be a good balance.

**Comment by bcpierce00** [comment]
I think simply making "camera / audio off" be the default when entering a room would be a big improvement -- maybe all we need.

**Comment by jon-bell** [comment]
FYI for ICSE we started muted by default in big group rooms and people hated it. If we do camera/audio off by default, we need to make it extremely obvious (much more so than how it is now) that this is the case and how to do it. It would also probably be good to make the video/audio mute sticky across calls (so if you turn video off in one room then jump into another, still off, if it's on in one then jump to another, still on). It's our oldest open bug :) #3 

 https://github.com/clowdr-app/clowdr-web-app/commit/b10348950ee9a30a163b02a276ea55b57dd75be8#diff-6f9cdfcc341448d4bf500001325d6e43

**Comment by EdNutting** [comment]
Sticky video/audio seems like the most effective solution to me. Perhaps with having two options for joining a room: per-previous (ordinary join) or silent (muted join).

**Comment by bcpierce00** [comment]
This is fine, but it doesn’t solve the original problem, namely that people often seem to feel shy about joining rooms where other “famous people” are already talking.  

My original plan was to enter rooms (by default, or perhaps as one of two options) in a “just listening” mode where your name (with no video or audio) is displayed in some special way that lets people know that you are just listening and where you are given a big clearly marked button that can be pushed to fully join the room.  I still think this is the most promising design to try.

**Comment by EdNutting** [comment]
I think that's a terrible over-complication - if people are one click away from joining the room properly using the "move from just listening to participation button", that's the same as the video/audio enable/disable buttons. 

I also think that labeling people in a separate section as "just listening" will have a significant negative impact: once someone is labeled as "just listening", it will raise the fear-level even higher of moving to contributing. In effect, by putting the label on screen for everyone to see (including the "famous people"), you increase the problem rather than decreasing it. This seems like a classic case of trying to use technology to solve a human problem: if people are afraid of joining rooms with "famous people" then that's a community problem (one which ICFP didn't seem to have, at least in my experience).

Also, at this stage, creating such a separate mechanism is going to be complex and costly compared to just having a button say "join muted" (which I think achieves the same result but without the stigmatizing label).

**Comment by EdNutting** [comment]
(The alternative is some kind of invisible observation mode - which is diabolical because then participants in the room can't tell if who's listening. People naturally become very closed and guarded if they think literally anyone could be listening and they don't know who.)

**Comment by EdNutting** [comment]
On a separate note: with a more stable video layout, entering a room with video/audio enabled would be much less disruptive as the videos wouldn't resize/move around/disappear.

**Comment by bcpierce00** [comment]
Yes, we don’t want this!

> (The alternative is some kind of invisible observation mode - which is diabolical because then participants in the room can't tell if who's listening. People naturally become very closed and guarded if they think literally anyone could be listening and they don't know who.

**Comment by bcpierce00** [comment]
> Also, at this stage, creating such a separate mechanism is going to be complex and costly compared to just having a button say "join muted" (which I think achieves the same result but without the stigmatizing label).
> 
I agree that this is the simplest way, and definitely worth doing.  From what people have said to me (yes, even at ICFP), I’m not certain it’s enough.  But it’s very possible that people can just be trained to equate “video/mic off” with “just listening."

**Comment by EdNutting** [comment]
Perhaps my bias is that I already equate "vid/mic off" with "just listening or AFK" - anyone used to remote working or online gaming is likely to already have this association (based on my experience and the experiences of my long-time remote working friends at other companies). Perhaps we just need to tell people this in a notice on screen, for those who are new to remote work / conferencing?

**Comment by jon-bell** [comment]
> On a separate note: with a more stable video layout, entering a room with video/audio enabled would be much less disruptive as the videos wouldn't resize/move around/disappear.

I think that this will solve a lot of our video chat usability problems. When you join a room, if you are video+audio on, you should show up either in the grid of people who have video on, or in a list (perhaps above that grid) that lets you select that person and see their video. If you have only audio on, or are fully muted, we should show you listed in some way to make extremely clear visually that you can not be clicked on to see your video (aka that you are "just listening).

**Comment by bcpierce00** [comment]
Maybe “extremely clear” = “labeled ‘just listening’” ?  :-)

**Comment by EdNutting** [comment]
I think we should follow Zoom's example: Layout everyone in a grid, try our best to always keep the positioning the same (boxes for participants are all equal size), and if someone turns off video, show their name in the box and if they mute, just add a muted symbol to their box. It's simple, effective, doesn't require lots of extra work, doesn't add unnecessary labels and will match user's experience on most other platforms (Zoom, MS Teams, Discord, etc) which will make it much easier for people to use our platform (there is no substitute for instant familiarity).

[Note: Account for 'presenter mode' separately]

**Comment by jon-bell** [comment]
@EdNutting the one issue is that we can't show more than 4 videos at a time (even zoom can't in their web app, only in native) - I think that having everyone in the grid sets people up to expect zoom's native experience (up to 40 or however many video streams), which then is immediately disappointing.

**Comment by bcpierce00** [comment]
Showing everyone all the time, even when A/V muted, also wastes screen real estate, which is a precious resource on some devices (not Jon’s :-)…

**Comment by EdNutting** [comment]
@jon-bell The above conversation does nothing to address that particular problem... none of the alternatives to a grid layout solve the issue of not showing more than 4 videos at once. The grid layout makes solving that one a lot easier if a stable layout is desired (which I think it absolutely should be): you simply disable the video and replace it with their name (or continue to try the freeze-frame thing but with something other than greying-out which makes people think of disconnection).

@bcpierce00 There is a tradeoff between stability of the layout and making sure the right person is always on screen. In practice, dropping to a 3x3 grid and a maximum room size of 9 would be perfectly acceptable even on small mobile devices (with at most 4 video feeds on at once).

**Comment by EdNutting** [comment]
(Or a 2x4 grid and a room size of 8 - depends a bit on which set of mobile devices you want to target - you can't win them all. Only fairly high end devices from the last 4 years are going to cope with a video calling app like this anyway)

**Comment by hannelita** [comment]
Would it make sense to have a 2-step join, just like in Slack? then the participants can choose to join as listeners; i.e. click on the room, see the bottom menu and then select the join mode
<img width="865" alt="Screen Shot 2020-09-01 at 11 13 32" src="https://user-images.githubusercontent.com/506426/91869853-5c60cb00-ec44-11ea-983d-dbc135ffa389.png">

**Comment by bcpierce00** [comment]
Yes, I think this would be fine.

**Comment by EdNutting** [comment]
This results in the silent observation problem mentioned above. It is acceptable in slack because the expectation is you know who is in the organisation. I don't think this is appropriate for a public conference environment. At a physical event, you wouldn't expect someone to be covertly listening in to a group conversation by standing on the other side of a room with a long range mic!

**Comment by bcpierce00** [comment]
Maybe I misunderstood the proposal.  I thought we were just talking about a two-step process to join where the first step was choosing between A/V off and A/V on.

**Comment by EdNutting** [comment]
Unless you meant not to show the room (video/audio/etc) until after the type of join has been selected? Which isn't like Slack's I don't think. That would be more like Zoom's behaviour.

**Comment by EdNutting** [comment]
@bcpierce00 Okay yeah that's Zoom's model not Slack's.
